app/controllers/catalog_controller.rb

```diff
@@ -11,6 +11,18 @@ class CatalogController < ApplicationController
     redirect_to error_page
   end
 
+  rescue_from Blacklight::Exceptions::ECONNREFUSED do
+    # If the Solr service is available, retry the HTTP request
+    if search_service.repository.ping
+      params.permit!
```

Comment:
I wonder if just issuing a `retry` here would make the spec pass.  It would be simpler code logic also see https://www.honeybadger.io/blog/how-to-try-again-when-exceptions-happen-in-ruby/